skops/io/_audit.py

```diff
@@ -269,15 +271,11 @@ def get_unsafe_set(self) -> set[str]:
                     if not check_type(get_module(child), child.__name__, self.trusted):
                         # if the child is a type, we check its safety
                         res.add(get_module(child) + "." + child.__name__)
-                elif isinstance(child, io.BytesIO):
+                elif isinstance(child, (io.BytesIO, str)):
                     # We trust BytesIO objects, which are read by other
-                    # libraries such as numpy, scipy.
-                    continue
-                elif check_type(
-                    get_module(child), child.__class__.__name__, PRIMITIVE_TYPE_NAMES
-                ):
```

Comment:
are we removing them cause now primitives are trusted by default?

Reply:
This is what I was referring to here:

> Another "victim" of this PR is that the so far dead code of checking for primitive types inside of `get_unsafe_set` has been removed. This code was supposed to check if the type is a primitive type but it was defective. `get_module(child)` would raise an error if an instance of the type would be passed. We could theoretically fix that code, but it would still be dead code because primitive types are stored as json.